pxd.c

```diff
@@ -1197,24 +1185,54 @@ static int pxd_init_disk(struct pxd_device *pxd_dev, struct pxd_add_ext_out *add
 	  pxd_dev->tag_set.cmd_size = sizeof(struct fuse_req);
 
 	  err = blk_mq_alloc_tag_set(&pxd_dev->tag_set);
-	  if (err)
-		goto out_disk;
+	  if (err) {
+		return err;
+	  }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,15,0)
+	  disk = blk_mq_alloc_disk(&pxd_dev->tag_set, pxd_dev);
+	  if (IS_ERR(disk)) {
+		blk_mq_free_tag_set(&pxd_dev->tag_set);
+		return PTR_ERR(disk);
+	  }
+	  q = disk->queue;
+#else
+	  /* Create gendisk info. */
```

Comment:
alloc_disk is called in both ifdef and ifndef PX_BLKMQ... can be taken out and if kernel version < 5,15,0 then can perform alloc_disk. outside above L1192.